evalml/utils/gen_utils.py

```diff
@@ -219,8 +219,16 @@ def _rename_column_names_to_numeric(X):
     Returns:
         Transformed X where column names are renamed to numerical values
     """
-    name_to_col_num = dict((col, col_num) for col_num, col in enumerate(X.columns.values))
-    return X.rename(columns=name_to_col_num, inplace=False)
+    X_t = X
+    if isinstance(X, np.ndarray):
+        return X
+    if isinstance(X, ww.DataTable):
+        X_t = X.to_dataframe()
+    name_to_col_num = dict((col, col_num) for col_num, col in enumerate(list(X.columns)))
+    X_renamed = X_t.rename(columns=name_to_col_num, inplace=False)
+    if isinstance(X, ww.DataTable):
+        return ww.DataTable(X_renamed)
```

Comment:
Ah, got it.

One point about this case: we should copy the types from the original woodwork datatable. These types could come from the default woodwork type inference, but they could also be user-defined. We need to preserve them. Otherwise, we could get in a situation where lightgbm/xgboost are fitting on a different set of categorical features for instance vs what the user specified.

@gsheni @thehomebrewnerd @tamargrey any advice on how we can do this?

@angela97lin we should add test coverage for this case too.

Reply:
DataTable has a `logical_types` property that's the same format as the `logical_types` dict you pass in on init, so you could grab that with `X.logical_types`, rename the columns along with the underlying data, and then pass it in on line 230. 

But as @freddyaboulton mentioned in another comment, DataTable has a `rename` method I think you could make use of here if you want

Reply:
Just confirmed with Tamar that the `rename` method is not in the latest release so sticking with this for now, but updating to make sure the logical types are carried over, and adding test coverage. Thanks all!